### PR TITLE
WL-378 Adding custom Django sites framework middleware to allow for a default site

### PR DESCRIPTION
**Pull request by douglashall**
Please see the thread on [edx-code](https://groups.google.com/forum/#!topic/edx-code/5q6riHK2_P8) for background on the need for this change.

@mattdrayer @justinabrahms @felipemontoya

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 .. |Travis| image:: https://travis-ci.org/edx/edx-django-sites-extensions.svg?branch=master
-.. _Travis: https://travis-ci.org/edx/edx-django-sites-extensions
+.. Travis: https://travis-ci.org/edx/edx-django-sites-extensions
 
 .. |Codecov| image:: http://codecov.io/github/edx/edx-django-sites-extensions/coverage.svg?branch=master
 .. Codecov: http://codecov.io/github/edx/edx-django-sites-extensions?branch=master
 
+
 This package includes extensions to the Django "sites" framework
 used by edX Django IDAs (independently deployable applications).
```
Comment (mattdrayer) [inline README.rst]: Do we need to say "Open edX" here?

File: django_sites_extensions/middleware.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""
+Django sites framework middleware extensions
```
Comment (mattdrayer) [inline django_sites_extensions/middleware.py]: I'd add "for Open edX" here

File: django_sites_extensions/middleware.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    def process_request(self, request):
+        try:
+            super(CurrentSiteWithDefaultMiddleware, self).process_request(request)
+        except Site.DoesNotExist:
+            site_id = getattr(settings, 'DEFAULT_SITE_ID', '')
+            if site_id:
+                request.site = Site.objects.get(pk=site_id)  # pylint: disable=no-member
+            else:
+                raise
```
Comment (mattdrayer) [inline django_sites_extensions/middleware.py]: Should we raise Site.DoesNotExist here, or ImproperlyConfigured?  I think the Django Sites framework raises ImproperlyConfigured, IIRC.

Comment (douglashall) [inline django_sites_extensions/middleware.py]: The sites framework raises `Site.DoesNotExist`. You can see this above where I am calling `process_request` on the parent middleware class.

Comment (mattdrayer) [inline django_sites_extensions/middleware.py]: Riiight :)  I feel like I saw ImproperlyConfigured somewhere...maybe it's not a big deal, tho

Comment (douglashall) [inline django_sites_extensions/middleware.py]: Yes, you could get an ImproperlyConfigured [here](https://github.com/django/django/blob/master/django/contrib/sites/models.py#L69), but we do not want to catch that one.

Comment (douglashall) [inline django_sites_extensions/middleware.py]: Scratch that. I think we do want to catch ImproperlyConfigured as well.

**Comment by mattdrayer** [comment]
PR LGTM now -- :+1:

**Pull request merged**